packages/layerchart/src/lib/components/ChartContext.svelte

```diff
@@ -86,9 +86,31 @@
 <script lang="ts" generics="TData">
   import type { SankeyGraph } from 'd3-sankey';
 
+  const dispatch = createEventDispatcher<{
+    resize: { width: number; height: number };
+  }>();
+
   const layerCakeContext = getContext<LayerCakeContext<TData>>('LayerCake');
   setChartContext(layerCakeContext);
 
+  const size = derived([layerCakeContext.width, layerCakeContext.height], ([$width, $height]) => ({
+    width: $width,
+    height: $height,
+  }));
```

Comment:
Not sure we need to add it now, but might be nice to also include `containerWidth` and `containerHeight` (full container size) whereas `width` and `height` exclude the `padding` size.

Reply:
Absolutely! Fixed.

I also got rid of the derived store altogether in favor of just accessing the individual stores within the `$: { … }`.

Also broke them up into `{ chart: { width, height }, container: { width, height } }` and exported their types.

I was contemplating also including a `{ …, padding: { left, right, bottom, top } }`, but there only seems to be `xPadding` and `yPadding`, so the padding can be inferred from `chart` + `container` due to axial symmetry.

Reply:
Regarding the event, I think we should stick with the same names as exposed as slot props (`<Chart ... let:width let:containerWidth>`) on the event, for example

```js
dispatch('resize', {
  width: $width,
  height: $height,
  containerWidth: $containerWidth,
  containerHeight: $containerHeight,
});
```

---

[xPadding](https://layercake.graphics/guide#xpadding) and [xPadding](https://layercake.graphics/guide#ypadding) are for padding the respective `xDomain` and `yDomain` (useful to give room within the chart area for labels above points, for example), while [padding](https://layercake.graphics/guide#padding) is used to shift the chart area within the container (useful to provide room for axises, legend, title, etc).

I personally think they have confusing names (maybe `xPadding` should have been `xDomainPadding`), and `padding` aligns with D3's "margin convention" but works more like CSS padding (inner) instead of margin (outer), hence the name.